packages/edit-site/src/components/save-hub/index.js

```diff
@@ -32,29 +55,106 @@ export default function SaveHub() {
 		[]
 	);
 
+	const {
+		editEntityRecord,
+		saveEditedEntityRecord,
+		__experimentalSaveSpecifiedEntityEdits: saveSpecifiedEntityEdits,
+	} = useDispatch( coreStore );
+
 	const disabled = isSaving || ( ! isDirty && ! isPreviewingTheme() );
 
+	// if we have only one unsaved change and it matches current context, we can show a more specific label
+	let dirtyLocal = null;
```

Comment:
I'm also thinking `dirtyLocal` should be set within the `useSelect` so that `useSelect` contains all the "logic" and just returns just the data needed to adjust the UI: `dirtyLocal`, `isDirty`, `isSaving`, and `countUnsavedChanges`.

Alternatively keep it as is but have `useSelect` return only `dirtyRecords` and `isSaving`.

It's just weird right now how `useSelect` returns both derived data (`isDirty`, `countUnsavedChanges`) and raw data (`dirtyRecords`).

Reply:
Nice idea, I've updated to include it in useSelect